### PR TITLE
Various tidy-ups and fixes

### PR DESCRIPTION
**Pull request by jhol**
This patch-set contains several useful improvements to Cannelloni:

  * An AutoTools-based build system.
  * A Nix flake definining the build configuration for the Nix package manager.
  * Various code tidy-ups.

### CODE REVIEW

**Comment by yomboprime** [comment]
> @yomboprime did you have a chance to look at this?

Oh sorry, didn't get an email. Thank you! will look into it.

File: README.md [hunk 1]
```diff
@@ -15,17 +15,15 @@ I get 36 MiB/s in/out with my 10-year old Intel i5 PC, using one of those FX2LP
 
 ## Installation
 
-Building the firmware is optional, it is already built in ```firmware/fx2pipe.ihx```. For building it, install sdcc first (it could be in your repositories). Then:
+Depedencies:
```
Comment (yomboprime) [inline README.md:18]: "Dependencies"

**Comment by jhol** [comment]
No problem. Thanks for reviewing the changes.

By the way, I am the main author of the fx2lafw logic analyzer firmware for sigrok, but I found myself needing fx2pipe for a work project. Thank you for reviving it.

I have quite a few other patches: a rewrite of the firmware to make it controllable through vendor commands rather than requiring binary patching, and some moves towards asynchronous USB transfers, but I wanted to begin by submitting some more foundational improvements.

**Comment by yomboprime** [comment]
Nice to know you have such improvements coming.

In 0.1.1 I implemented (besides fixing some bugs you've found) an optional transfer mode with variable block size, where each block is preceded by a 32 bit word with the block length (though the block length word is not transferred to usb) I hope this mode is compatible with your changes.

I have Sigrok installed and was planning to make a logic analizer with a FX2 and use it some day.

**Comment by yomboprime** [comment]
Again, thanks for this. Please wait for a series of commits I've got before doing more PR's. Probably I will finish today.

**Pull request merged**